homeassistant/components/calendar/caldav.py

```diff
@@ -131,6 +135,36 @@ def __init__(self, calendar, include_all_day, search):
         self.search = search
         self.event = None
 
+    async def async_get_events(self, start_date, end_date):
+        """Get all events in a specific time frame."""
+        # Get event list from the current calendar
+        vevent_list = self.calendar.date_search(start_date, end_date)
```

Comment:
Is this doing any I/O or is this operating on an in-memory structure ?

If it's doing I/O, run it in the thread pool

```python
await self.hass.async_add_job(self.calendar.date_search, start_date, end_date)
```

(you might have to pass in `hass` if it doesn't have access to it.)